### PR TITLE
SSL Listener support

### PR DESCRIPTION
**Pull request by mdurrant-b3**
Adds listeners if they're specified, additional NotFound error number because we don't have enough of those.

### CODE REVIEW

**Comment by dougwig** [comment]
pep8 failure

**Comment by Cedev** [comment]
We should probably add the template support for 2.1 at some point.

**Comment by dougwig** [comment]
We either need not implemented handlers or the real code in 2.1.

**Comment by Cedev** [comment]
Tested in 3.0, works nicely.

**Comment by Cedev** [comment]
Virtual server port creation is implemented in 2.1, it doesn't set the ssl templates on the port that are now passed as arguments to port create. `port.create(..., 80, 'https', ...,  template_client_ssl=X)` is equivalent to

```
    port 80 https
        ...
        template client-ssl X
```

tl;dr. It's an attribute, there's no method to be NotImplemented

**Comment by mdurrant-b3** [comment]
v21 support WIP.
